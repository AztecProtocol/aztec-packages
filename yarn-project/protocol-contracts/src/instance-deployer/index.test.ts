import omit from 'lodash.omit';

import { InstanceDeployerAddress, getCanonicalInstanceDeployer } from './index.js';

describe('InstanceDeployer', () => {
  it('returns canonical protocol contract', () => {
    // TODO(@spalladino): Consider sorting functions by selector when constructing the contract
    // class, or even better, when calling loadContractArtifact from the Noir output.
    const contract = getCanonicalInstanceDeployer();
    contract.contractClass.privateFunctions.sort((a, b) => a.selector.value - b.selector.value);
    contract.contractClass.publicFunctions.sort((a, b) => a.selector.value - b.selector.value);
    expect(omit(contract, 'artifact')).toMatchSnapshot();
    expect(contract.address.toString()).toEqual(InstanceDeployerAddress.toString());
  });
});
