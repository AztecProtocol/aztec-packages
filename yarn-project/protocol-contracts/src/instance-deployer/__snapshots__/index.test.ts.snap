// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceDeployer returns canonical protocol contract 1`] = `
{
  "address": AztecAddress {
    "asBigInt": 3292754564016949916718877297115777598905061669834342030471417311460989046852n,
    "asBuffer": {
      "data": [
        7,
        71,
        162,
        14,
        208,
        200,
        96,
        53,
        228,
        78,
        165,
        96,
        111,
        48,
        222,
        69,
        159,
        64,
        181,
        92,
        94,
        130,
        1,
        38,
        64,
        170,
        85,
        69,
        70,
        175,
        144,
        68,
      ],
      "type": "Buffer",
    },
  },
  "contractClass": {
    "artifactHash": Fr {
      "asBigInt": 14004623793815794786663172306952806542169300491436089583084017080489106515164n,
      "asBuffer": {
        "data": [
          30,
          246,
          86,
          15,
          36,
          14,
          183,
          120,
          232,
          198,
          163,
          244,
          175,
          96,
          159,
          150,
          221,
          22,
          123,
          72,
          71,
          20,
          2,
          134,
          20,
          176,
          2,
          23,
          9,
          248,
          164,
          220,
        ],
        "type": "Buffer",
      },
    },
    "id": Fr {
      "asBigInt": 18800681354736902372207149850261995699819228885238981092036778984892317144615n,
      "asBuffer": {
        "data": [
          41,
          144,
          206,
          253,
          46,
          88,
          237,
          15,
          191,
          185,
          124,
          36,
          46,
          254,
          144,
          100,
          66,
          216,
          236,
          181,
          113,
          20,
          163,
          159,
          75,
          11,
          21,
          162,
          80,
          133,
          150,
          39,
        ],
        "type": "Buffer",
      },
    },
    "packedBytecode": {
      "data": [
        0,
        0,
        0,
        0,
      ],
      "type": "Buffer",
    },
    "privateFunctions": [
      {
        "isInternal": false,
        "selector": FunctionSelector {
          "value": 2285065643,
        },
        "vkHash": Fr {
          "asBigInt": 0n,
          "asBuffer": {
            "data": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "type": "Buffer",
          },
        },
      },
      {
        "isInternal": false,
        "selector": FunctionSelector {
          "value": 2432309179,
        },
        "vkHash": Fr {
          "asBigInt": 0n,
          "asBuffer": {
            "data": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "type": "Buffer",
          },
        },
      },
    ],
    "privateFunctionsRoot": Fr {
      "asBigInt": 10175937322611504760053163396285173722489074882473725995897130952520062304283n,
      "asBuffer": {
        "data": [
          22,
          127,
          96,
          93,
          15,
          118,
          143,
          145,
          206,
          97,
          129,
          56,
          23,
          230,
          12,
          1,
          170,
          115,
          219,
          197,
          231,
          133,
          116,
          128,
          74,
          71,
          30,
          51,
          115,
          5,
          128,
          27,
        ],
        "type": "Buffer",
      },
    },
    "publicBytecodeCommitment": Fr {
      "asBigInt": 13424778679590722659186070167781649503057405647123863047387123265455936508181n,
      "asBuffer": {
        "data": [
          29,
          174,
          39,
          204,
          127,
          226,
          175,
          52,
          95,
          22,
          2,
          83,
          190,
          56,
          117,
          212,
          73,
          167,
          233,
          186,
          107,
          214,
          135,
          71,
          216,
          125,
          78,
          112,
          84,
          184,
          17,
          21,
        ],
        "type": "Buffer",
      },
    },
    "publicFunctions": [],
    "version": 1,
  },
  "instance": {
    "address": AztecAddress {
      "asBigInt": 3292754564016949916718877297115777598905061669834342030471417311460989046852n,
      "asBuffer": {
        "data": [
          7,
          71,
          162,
          14,
          208,
          200,
          96,
          53,
          228,
          78,
          165,
          96,
          111,
          48,
          222,
          69,
          159,
          64,
          181,
          92,
          94,
          130,
          1,
          38,
          64,
          170,
          85,
          69,
          70,
          175,
          144,
          68,
        ],
        "type": "Buffer",
      },
    },
    "contractClassId": Fr {
      "asBigInt": 18800681354736902372207149850261995699819228885238981092036778984892317144615n,
      "asBuffer": {
        "data": [
          41,
          144,
          206,
          253,
          46,
          88,
          237,
          15,
          191,
          185,
          124,
          36,
          46,
          254,
          144,
          100,
          66,
          216,
          236,
          181,
          113,
          20,
          163,
          159,
          75,
          11,
          21,
          162,
          80,
          133,
          150,
          39,
        ],
        "type": "Buffer",
      },
    },
    "initializationHash": Fr {
      "asBigInt": 5411687428194306823799197943510367074755080310726877398642044502900791103391n,
      "asBuffer": {
        "data": [
          11,
          246,
          232,
          18,
          241,
          75,
          176,
          41,
          247,
          203,
          156,
          141,
          168,
          54,
          125,
          217,
          124,
          6,
          142,
          120,
          141,
          79,
          33,
          0,
          127,
          217,
          112,
          20,
          235,
          168,
          207,
          159,
        ],
        "type": "Buffer",
      },
    },
    "portalContractAddress": EthAddress {
      "buffer": {
        "data": [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
        ],
        "type": "Buffer",
      },
    },
    "publicKeysHash": Fr {
      "asBigInt": 17954372136461894568820199868962143715279932924957662285438325406238856880109n,
      "asBuffer": {
        "data": [
          39,
          177,
          208,
          131,
          154,
          91,
          35,
          186,
          241,
          42,
          141,
          25,
          91,
          24,
          172,
          40,
          143,
          207,
          64,
          26,
          251,
          47,
          112,
          184,
          164,
          181,
          41,
          237,
          229,
          250,
          159,
          237,
        ],
        "type": "Buffer",
      },
    },
    "salt": Fr {
      "asBigInt": 1n,
      "asBuffer": {
        "data": [
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          0,
          1,
        ],
        "type": "Buffer",
      },
    },
    "version": 1,
  },
}
`;
